Tests/Realm.Tests/Helpers/RequiresBaasAttribute.cs

```diff
@@ -0,0 +1,94 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Realms.Tests.Sync
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    internal class RequiresBaasAttribute : TestActionAttribute, IApplyToTest, IApplyToContext
+    {
+        private const string ExceptionKey = "Realm_SyncSession_Exception";
+        private const string ErrorHandlerKey = "Realm_SyncSession_ErrorHandler";
+
+        public bool EnsureNoSessionErrors { get; set; }
+
+        public override ActionTargets Targets => ActionTargets.Test;
+
+        public void ApplyToTest(Test test)
+        {
+            if (!SyncTestHelpers.HasBaas)
+            {
+                test.RunState = RunState.Skipped;
+                test.Properties.Set(PropertyNames.SkipReason, "MongoDB Realm is not set up.");
+            }
+        }
+
+#if UNITY
+        public void ApplyToContext(ITestExecutionContext context)
+#else
+        public void ApplyToContext(TestExecutionContext context)
+#endif
+        {
+            if (context.TestCaseTimeout == 0)
+            {
+                // apply the default timeout for Baas tests if no timeout was defined
+                context.TestCaseTimeout = 30000;
+            }
+        }
+
+        public override void BeforeTest(ITest test)
+        {
+            Task.Run(() => SyncTestHelpers.CreateBaasAppsAsync()).Wait();
+
+            if (EnsureNoSessionErrors)
+            {
+                void ErrorHandler(object _, ErrorEventArgs e)
+                {
+                    test.Properties.Set(ExceptionKey, e.Exception);
+                }
+
+                var @delegate = new EventHandler<ErrorEventArgs>(ErrorHandler);
+                Realms.Sync.Session.Error += @delegate;
```

Comment:
As we discussed already, we'll need to re-adjust this once the client reset API PR gets merged as the `Session.Error` is deprecated from there on